### DISCUSSION TITLE
Is creating a BBCode extension possible?

### DISCUSSION THREAD

**Discussion by Art4** [thephpleague/commonmark#755] [Q&A]
Hey, I have a legacy project with user generated content written in BBCode. I would like to migrate this BBCode content to CommonMark and have started to build a BBCode extension. See here for my progress: https://github.com/youthweb/bbcode-parser/pull/15

Now comes the first obstacle for creating a Delimiter for detecting `[b]text[/b]` and parsing it to `<strong>text</strong>`. I have started looking into InlineParser and BlockPArser but my question is: Is it generally possible to detect BBCode with league/commonmark and what are the right tools for detecting simple BBCode rules?

Or more specific: How can I parse `Hello [b]World[/b]!` into `<p>Hello <strong>World</strong>!</p>`? Should I create a BlockParser or a InlineParser?

**Comment by leevigraham** [comment]
@Art4 In the same boat… did you solve this?

**Reply by Art4** [reply]
Not yet. My progress stucked because of my last question. My progress so far is linked above.

**Reply by kingRayhan** [reply]
I am also looking for this :cry:

**Comment by colinodell** [comment]
> Is it generally possible to detect BBCode with league/commonmark

_Probably_.  I'm not very familiar with BBCode, so I don't know if you could properly implement everything, but parsing the type of syntax you showed would probably work.

> How can I parse Hello [b]World[/b]! into <p>Hello <strong>World</strong>!</p>? Should I create a BlockParser or a InlineParser?

You would use an `InlineParser` for this.  See https://spec.commonmark.org/0.30/#blocks-and-inlines to better understand the difference between the two.  Generally speaking, anything that would be considered an inline element [in HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Inline_elements) would also be considered an inline in CommonMark.
